lib/msf/core/modules/metadata/obj.rb

```diff
@@ -65,6 +111,44 @@ def path
 
     @path
   end
+
+  #######
+  private
+  #######
+
+  def init_from_hash(obj_hash)
+    @name = obj_hash['name']
+    @full_name = obj_hash['full_name']
+    @disclosure_date = obj_hash['disclosure_date'].nil? ? nil : Time.parse(obj_hash['disclosure_date'])
+    @rank = obj_hash['rank']
+    @type = obj_hash['type']
+    @description = obj_hash['description']
+    @author = obj_hash['author'].nil? ? [] : obj_hash['author']
+    @references = obj_hash['references']
+    @is_server = obj_hash['is_server']
+    @is_client = obj_hash['is_client']
+    @platform = obj_hash['platform']
+    @arch = obj_hash['arch']
+    @rport = obj_hash['rport']
+    @mod_time = Time.parse(obj_hash['mod_time'])
+    @ref_name = obj_hash['ref_name']
+    @path = obj_hash['path']
+    @is_install_path = obj_hash['is_install_path']
+    @targets = obj_hash['targets'].nil? ? [] : obj_hash['targets']
+  end
+
+  def sort_platform_string
+    arr = @platform.split(',')
+    if !arr.nil? && arr.length > 1
```

Comment:
`arr` is initialised with a call to `split` on the previous line. A call to `split` should never return `nil`, so `arr` should never be `nil`, making the `nil` check redundant.

If the `split` call fails, you have bigger problems.

Also, `size` is preferred over `length` for array length comparison, but it's a [suggestion](https://github.com/rubocop-hq/ruby-style-guide#map-find-select-reduce-size), not a requirement.


```ruby
  def sort_platform_string
    arr = @platform.split(',')
    @platform = arr.sort.join(',') if arr.size > 1
  end
```

Reply:
thanks